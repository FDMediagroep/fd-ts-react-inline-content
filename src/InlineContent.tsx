import {BulletPoints, BulletPointsStyle} from './components/BulletPoints';
import {LinkBlock, LinkBlockStyle} from './components/LinkBlock';
import {NumberFrame, NumberFrameStyle} from './components/NumberFrame';
import {Quote, QuoteStyle} from './components/Quote';
import {StackFrame, StackFrameStyle, StackFrameDataType} from './components/StackFrame';
import {Stock, StockStyle} from './components/Stock';
import {Summary, SummaryStyle} from './components/Summary';
import {TextFrame, TextFrameStyle} from './components/TextFrame';

export {BulletPoints, BulletPointsStyle};
export {LinkBlock, LinkBlockStyle};
export {NumberFrame, NumberFrameStyle};
export {Quote, QuoteStyle};
export {StackFrame, StackFrameStyle, StackFrameDataType};
export {Stock, StockStyle};
export {Summary, SummaryStyle};
export {TextFrame, TextFrameStyle};
