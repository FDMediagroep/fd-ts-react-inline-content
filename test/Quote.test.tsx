import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Quote } from '../src/InlineContent';

describe('Quote', ()  => {
    const quote = "Curabitur quis luctus est, vitae sodales enim";
    const figcaption = "Willem Liu";

    it('should render align right correctly', () => {
        let component = TestRenderer.create(<Quote {...{quote, figcaption}} />);
        expect(component.toJSON()).toMatchSnapshot();

        component = TestRenderer.create(<Quote {...{quote, figcaption}} align='right'/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render align block correctly', () => {
        let component = TestRenderer.create(<Quote {...{quote, figcaption}} align='block'/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

});
