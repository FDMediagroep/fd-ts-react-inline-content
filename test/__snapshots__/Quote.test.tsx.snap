// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should render align block correctly 1`] = `
<figure
  className="fd-typography article-inline-quote inline-content quote block"
  textStyle="article-inline-quote"
>
  <blockquote>
    ‘
    Curabitur quis luctus est, vitae sodales enim
    ’
  </blockquote>
  <figcaption>
    • 
    Willem Liu
  </figcaption>
</figure>
`;

exports[`Quote should render align right correctly 1`] = `
<figure
  className="fd-typography article-inline-quote inline-content quote right"
  textStyle="article-inline-quote"
>
  <blockquote>
    ‘
    Curabitur quis luctus est, vitae sodales enim
    ’
  </blockquote>
  <figcaption>
    • 
    Willem Liu
  </figcaption>
</figure>
`;

exports[`Quote should render align right correctly 2`] = `
<figure
  className="fd-typography article-inline-quote inline-content quote right"
  textStyle="article-inline-quote"
>
  <blockquote>
    ‘
    Curabitur quis luctus est, vitae sodales enim
    ’
  </blockquote>
  <figcaption>
    • 
    Willem Liu
  </figcaption>
</figure>
`;
